src/App.js

```diff
@@ -1,20 +1,441 @@
+     * componentDidMount() is used to retrieve data from 'localStorage' when the component, App.js, is mounted for the first time or refreshed in the browser
+     */
+    const savedTasks = localStorage.getItem("tasks");
+    const savedTasksInProgress = localStorage.getItem("tasksInProgress");
+    const savedTasksInReview = localStorage.getItem("tasksInReview");
+    const savedTasksCompleted = localStorage.getItem("tasksCompleted");
+
+    if (savedTasks) {
+      this.setState({
+        tasks: JSON.parse(savedTasks),
+        tasksInProgress: JSON.parse(savedTasksInProgress) || [],
+        tasksInReview: JSON.parse(savedTasksInReview),
+        tasksCompleted: JSON.parse(savedTasksCompleted),
+      });
+    }
+  }
+
+  // TASKS IN PROGRESS SECTION
+  updateTaskInProgress = (id, updatedTask) => {
+    this.setState((prevState) => {
+      const updatedTasks = prevState.tasksInProgress.map((task) => {
+        if (task.id === id) {
+          return { ...task, ...updatedTask };
+        }
+        return task;
+      });
+      return { tasksInProgress: updatedTasks };
+    }, this.updateLocalStorage);
+  };
+
+  addTaskToDoInProgress = (task) => {
+    const newId = this.state.tasksInProgress.length;
+    const newTask = { ...task, id: newId };
+    this.setState(
+      (prevState) => ({
+        tasksInProgress: [...prevState.tasksInProgress, newTask],
+      }),
+      this.updateLocalStorage
+    );
+  };
+
+  deleteTaskInProgress = (id) => {
+    this.setState(
+      (prevState) => ({
+        tasksInProgress: prevState.tasksInProgress.filter(
+          (task) => task.id !== id
+        ),
+      }),
+      this.updateLocalStorage
+    );
+  };
+
+  resetLocalStorageInProgress = () => {
+    localStorage.clear();
+    this.setState(
+      {
+        tasksInProgress: [
+          // {
+          //   id: 0,
+          //   title: "React Website - ORION Project",
+          //   task: "Coding a frontend website for Doctor Halsey",
+          // },
+        ],
+      },
+      this.updateLocalStorage
+    );
+  };
+
+  // TASKS IN REVIEW
+  updateTaskInReview = (id, updatedTask) => {
+    this.setState((prevState) => {
+      const updatedTasks = prevState.tasksInReview.map((task) => {
+        if (task.id === id) {
+          return { ...task, ...updatedTask };
+        }
+        return task;
+      });
+      return { tasksInReview: updatedTasks };
+    }, this.updateLocalStorage);
+  };
+
+  addTaskToDoInReview = (task) => {
+    const newId = this.state.tasksInReview.length;
+    const newTask = { ...task, id: newId };
+    this.setState(
+      (prevState) => ({ tasksInReview: [...prevState.tasksInReview, newTask] }),
+      this.updateLocalStorage
+    );
+  };
+
+  deleteTaskInReview = (id) => {
+    this.setState(
+      (prevState) => ({
+        tasksInReview: prevState.tasksInReview.filter((task) => task.id !== id),
+      }),
+      this.updateLocalStorage
+    );
+  };
+
+  resetLocalStorageInReview = () => {
```

Comment:
what's the difference between all these reset functions? You always clear the whole local storage and you reset a single item in state. What is the reason for that? If you could clear only a single key on local storage I would somewhat be able to follow your train of thought here.

Reply:
if we have the status passed as argument we can do sth like:

``` 
resetTaskStatus = (status) => {
     localStorage.clear();
     this.setState(
     {  [status]: [] }
     )
}
```